### PR TITLE
Adding system health checks to probe processes.

### PR DESCRIPTION
**Pull request by mangai90**

### CODE REVIEW

**Comment by rgooch** [comment]
Calling the package "process" may conflict with a possible later package which does process scanning. How about calling it "pidfile"?

File: cmd/health-agent/main.go [hunk 1]
```diff
@@ -14,6 +14,8 @@ import (
 )
 
 var (
+	configDir = flag.String("configDir", "/etc/health-agent/tests.d/",
```
Comment (rgooch) [inline cmd/health-agent/main.go]: Please make the default be "/etc/health-agent" and then internally add the "tests.d"

File: probers/pidfile/api.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/Symantec/tricorder/go/tricorder"
+	"os"
+)
+
+type pidconfig struct {
+	processname   string
+	pidfilepath   string
+	process       os.Process
+	healthy       bool
+	pidfileexists bool
```
Comment (rgooch) [inline probers/pidfile/api.go]: Please name these "pidFileExists" and "pidExists".

File: probers/pidfile/probe.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import (
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+func (p *pidconfig) probe() error {
+	p.setpidfile(true)
```
Comment (rgooch) [inline probers/pidfile/probe.go]: Why is this line here? This will set a transient "healthy" for an unhealthy service. This should be at the end of the function.

File: probers/pidfile/api.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+import (
+	"github.com/Symantec/tricorder/go/tricorder"
+	"os"
+)
+
+type pidconfig struct {
+	processname   string
+	pidfilepath   string
+	process       os.Process
```
Comment (rgooch) [inline probers/pidfile/api.go]: This does not seem to be used to preserve useful state between calls to Probe(), so it's simpler and cleaner to just create a local variable in the code that uses it.

**Comment by mangai90** [comment]
Committed the changes. PTAL.
